compiler/src/dotty/tools/dotc/ast/TreeInfo.scala

```diff
@@ -659,10 +659,8 @@ trait TypedTreeInfo extends TreeInfo[Type] { self: Trees.Instance[Type] =>
           rname == tree.name || hasRefinement(parent)
         case tp: TypeProxy =>
           hasRefinement(tp.underlying)
-        case tp: OrType =>
+        case tp: AndOrType =>
```

Comment:
Am I correct in assuming that if an intersection type has a component `C` that has a (non-structural) member `f` then a selection of `.f` on a value of that type will have its symbol filled in before `isStructuralTermSelect` is called? 
Thus the condition `!tree.symbol.exists` below will not be satisfied, and we will not be issuing unnecessary structural selections. This is even if the type intersection also has a structural type containing a field `f` (on the other hand it seems like such types are consistently converted into refinements on `C`, e.g. `C & {f:A}` is represented as `C{f:A}` or just `C` if `C#f =:= A`).